src/bp-groups/bp-groups-functions.php

```diff
@@ -2969,7 +2969,7 @@ function bp_groups_get_group_types( $args = array(), $output = 'names', $operato
 		$types = bp_get_taxonomy_types( bp_get_group_type_tax_name(), $types );
 	}
 
-	$types = wp_filter_object_list( $types, $args, $operator );
+	$types = array_filter( wp_filter_object_list( $types, $args, $operator ) );
```

Comment:
This concern is related to all those array_filter in this pull request.

Should we run `array_values` as well?

> Array keys are preserved, and may result in gaps if the array was indexed. The result array can be reindexed using the [array_values()](https://www.php.net/manual/en/function.array-values.php) function.

Source: https://www.php.net/manual/en/function.array-filter.php

Reply:
I don't think so. I'm using it to remove empty values from the array.

Reply:
That's correct but if those values were an array with indexes, the result array index will not be the same. We'd be essentially change the structure of the array.

Reply:
Right, but I prefer to preserve keys than losing them in favor of a numerically indexed array

> `array_values()` returns all the values from the array and indexes the array **numerically**.

Reply:
🤔 

Maybe what I'm saying is that you are **losing the order** of the index using `array_filter`, which might be unexpected?! Or is that expected?

Reply:
I'm unsure these arrays are numerically indexed. I'm beat for tonight! I need some rest. WP 5.7 is still problematic: unit tests are running with PHPUnit 7.5 but are failing 😢 . Let's look at it later!

Reply:
If they are not numerically indexed, this suggestion can be skipped. Just wanted to raise that to make sure. :)